### PR TITLE
New types v3

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
cc @EPashkin @sdroege

### CODE REVIEW

File: src/auto/bytes_icon.rs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            from_glib_full(ffi::g_bytes_icon_new(bytes.to_glib_none().0))
+        }
+    }
+}
+
+pub trait BytesIconExt {
+    #[cfg(any(feature = "v2_38", feature = "dox"))]
+    fn get_bytes(&self) -> Option<glib::Bytes>;
+
+    fn get_property_bytes(&self) -> Option<glib::Bytes>;
```
Comment (EPashkin) [inline src/auto/bytes_icon.rs]: This property need version 2_38 to be hidden

File: src/lib.rs [hunk 2]
```diff
@@ -1,8 +1,9 @@
-// Copyright 2013-2016, The Gtk-rs Project Developers.
+// Copyright 2013-2018, The Gtk-rs Project Developers.
 // See the COPYRIGHT file at the top-level directory of this distribution.
 // Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
 
 #![allow(deprecated)]
+#![feature(extern_prelude)]
```
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply src/lib.rs]: Apparently not. But without it, it doesn't work on nightly anymore. I'll need a new cfg, yeay, breaking changes...

[the remark this answers is not in the document]
Comment (EPashkin) [reply src/lib.rs]: Its gir error, it don't add `use glib;` to `auto\loadable_icon.rs`.

File: src/auto/bytes_icon.rs (hunk 1 above)
Comment (GuillaumeGomez) [reply src/auto/bytes_icon.rs]: Why? https://developer.gnome.org/gobject/stable/gobject-The-Base-Object-Type.html#g-object-get-property There is no minimum version...

Comment (EPashkin) [reply src/auto/bytes_icon.rs]: It minimum version for class itself and we already have `https://developer.gnome.org/gio/stable/GBytesIcon.html#g-bytes-icon-get-bytes`, so better hide property by setting same version to it.
Maybe this need do gir automatically.

PS. Strange that no set property generated

Comment (EPashkin) [reply src/auto/bytes_icon.rs]: Ah understand why no set, it Write/Construct only.

Comment (GuillaumeGomez) [reply src/auto/bytes_icon.rs]: Ok, I add the version then.

**Comment by GuillaumeGomez** [comment]
@EPashkin: Like this?

**Comment by EPashkin** [comment]
Yes, all looks good. Thanks.

**Comment by GuillaumeGomez** [comment]
Waiting for @sdroege now. :)

**Comment by sdroege** [comment]
Looks good to me

**Pull request merged**